Adjusted IPv6 test to perform when only valid mmask6

dmurphy18 opened the pull request:
### What does this PR do?

Fixes test for valid Ipv6 packet to only occur if IPv6 mmask6 is valid
### What issues does this PR fix or reference?

On AIX IPv6 packets were causing KeyErrors since 'prefixlen' index was being utilized without it having been defined. Utilizing the test only when mmask6 is valid ensures that 'prefixlen' index is defined prior to the test.  For example: line inet6 ::1%1/0 generates a mmask6 value of NoneType.
### Previous Behavior

KeyError due to use of undefined index 'prefixlen' with some IPv6 packets.
### New Behavior

Remove cause of KeyError by only testing after 'prefixlen' index has been defined
### Tests written?

No, debugged on AIX

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Commit:
cache query args with url as well

Without this, we can't use query args to get the hash of a file or add
query args to change what a file returns

puneetk committed:
No force_yes parameter to pkg.upgrade #21248 

We have a case for force_yes and skip_verify when upgrade

puneetk committed:
Fix pylint whitespace errors

puneetk committed:
Update aptpkg.py

puneetk committed:
Update aptpkg.py

Commit:
Adjusted IPv6 test to perform when only valid mmask6

Commit:
Adjusted IPv6 test to perform when only valid mmask6

Commit:
Update versionadded and release notes (#36352)

twangboy committed:
Add '/y' switch to the net stop and start commands

twangboy committed:
Add windows requirements file

xiaoanyunfei committed:
fix salt-api log and pid

Commit:
Merge pull request #36384 from twangboy/update_setup_req

Add windows requirements file

cachedout reviewed (commented):
Given this logic, how does the `inet6` key in the `data` dictionary get populated for valid addresses which are not masks?

Commit:
Merge pull request #36381 from twangboy/fix_win_service

Add '/y' switch to the net stop and start commands

Commit:
Merge pull request #36315 from puneetk/patch-6

No force_yes parameter to pkg.upgrade #21248

Commit:
Lint

Commit:
Merge pull request #36389 from cachedout/pr-36386

Pr 36386

Commit:
git.latest: fail gracefully for misconfigured remote repo (#36391)

* git.latest: fail gracefully for misconfigured remote repo

When the remote repo's HEAD refers to a nonexistent ref, this was
causing a traceback when we tried to check if the upstream tracking
branch needed to be changed after cloning the repo. This commit fixes
this traceback by gracefully failing the state when the remote HEAD is
not present in the ``git ls-remote`` output, but the desired remote
revision doesn't exist.

Additionally, a similar graceful failure now happens if the state is run
again after we gracefully fail the first time, and we need to set the
tracking branch. Trying to set the tracking branch when there is no
local branch would fail with an ambiguous error like "fatal: branch
'master' does not exist", so before we even attempt to set the tracking
branch, the state is failed with a more descriptive comment.

* Add integration test for #36242

dmurphy18 commented:
@cachedout actually digging further, every valid IPv6 packet should have a mmask6 value, given the search value from pmask6. So we should be okay in checking for a valid mmask6.  The AIX line is 'inet6 ::1%1/0', and that '/0' should be giving the mask but for the '%1' is messing up our search. I believe the '%1' could be an AIX issue on internal interface 'lo0' and we are not doing sufficient IPv6 testing to check, however the original code has paths which can use an undefined index which the PR fixes. 

Going to test on Linode with IPv6 to double-check IPv6 handling.

Commit:
Fix OS identification for CloudLinux (#36408)

jacobhammons committed:
Fix typo (#36409)

Commit:
Merge branch '2015.8' into '2016.3'

No conflicts.

thatch45 committed:
Merge pull request #36305 from gtmanfred/2016.3

cache query args with url as well

Commit:
Merge pull request #36413 from rallytime/merge-2016.3

[2016.3] Merge forward from 2015.8 to 2016.3

Commit:
Update to working AIX patch due to PR comments

Commit:
Merge branch 'aix_if_fix' of https://github.com/dmurphy18/salt into aix_if_fix
